pkg/gui/quitting.go

```diff
@@ -29,6 +29,18 @@ func (gui *Gui) handleQuitWithoutChangingDirectory() error {
 	return gui.quit()
 }
 
+func (gui *Gui) toggleWhitespaceInDiffView() error {
+	gui.State.IgnoreWhitespaceInDiffView = !gui.State.IgnoreWhitespaceInDiffView
+	var toastMessage string
+	if gui.State.IgnoreWhitespaceInDiffView {
+		toastMessage = gui.Tr.IgnoringWhitespaceInDiffView
+	} else {
+		toastMessage = gui.Tr.ShowingWhitespaceInDiffView
+	}
```

Comment:
100% nit pick: consider the below instead of using an `else` (cleaner imo):

```go
toastMessage := gui.Tr.ShowingWhitespaceInDiffView
if gui.State.IgnoreWhitespaceInDiffView {
	toastMessage = gui.Tr.IgnoringWhitespaceInDiffView
}
```

Reply:
ok; i literally copied from @jesseduffield but can change.

have i mentioned that my go-fu is weak?

Reply:
@harrisonhjones perhaps you can assist with the questions I have above tho?

Reply:
@harrisonhjones if I implement your suggestion:

```golang

func (gui *Gui) toggleWhitespaceInDiffView() error {
	gui.State.IgnoreWhitespaceInDiffView = !gui.State.IgnoreWhitespaceInDiffView

	toastMessage := gui.Tr.ShowingWhitespaceInDiffView
	if gui.State.IgnoreWhitespaceInDiffView {
		toastMessage := gui.Tr.IgnoringWhitespaceInDiffView
	}
	gui.raiseToast(toastMessage)

	return gui.refreshFilesAndSubmodules()
}
```

then when I run `go build`, go complains that `toastMessage declared but not used` and the line number given is the line inside the `if`, so I assume there's something I don't understand about scoping here (ie that the variable inside the `if` shadows the outer one and is being discarded). What am I missing?

Reply:
> ok; I literally copied from @jesseduffield but can change.
> 
> Have I mentioned that my go-fu is weak? In C#, TS or JS, I would have preferred a terniary here, tbh. In rust (and I'm new there too), it would have also, iirc, been a "one-liner". I don't know what the community standards are for go - generally, I like to stick with whatever is the most generally-accepted style for a community.

I know! That's why I said it was a nit pick. I can't find the reference but afaik Go community prefer to avoid else if possible.

Reply:
> @harrisonhjones perhaps you can assist with the questions I have above tho?

Let me take a look.

Reply:
@harrisonhjones for the record I consider the approach with `var toastMessage string` more expressive, though due to lazyness I typically go with your approach. I think that declaring the variable and then assigning the variable means you're not confusing the reader by making them think we're assigning one value, only to then assign another value. Of course, I would prefer if `if` statements in go were themselves expressions, or if we had ternaries, so that I didn't need to weigh up between these two options in the first place haha